capplets/common/mate-theme-apply.c

```diff
@@ -48,13 +50,24 @@ void
 mate_meta_theme_set (MateThemeMetaInfo *meta_theme_info)
 {
   GSettings *interface_settings;
+  GSettings *interface_gnome_settings;
   GSettings *marco_settings;
   GSettings *mouse_settings;
   GSettings *notification_settings = NULL;
   gchar *old_key;
   gint old_key_int;
+  gboolean wayland;
+
+  /*Find out whether we are running under wayland or x11 */
+
+  if (GDK_IS_X11_DISPLAY ((gdk_display_get_default())))
+    wayland = FALSE;
+
+  else
+    wayland = TRUE;
```

Comment:
What about doing something like this instead?
```c
    if (GDK_IS_X11_DISPLAY ((gdk_display_get_default())))
        interface_gnome_settings = g_settings_new (INTERFACE_GNOME_SCHEMA);
    else
        interface_gnome_settings = NULL;

/* … */

    if (interface_gnome_settings)
        g_settings_set_string (interface_gnome_settings, 
                               GTK_THEME_KEY, meta_theme_info->gtk_theme_name);
```
Just to avoid coupling `wayland` and `interface_gnome_settings`.

Just an idea, feel free to scrap it :)

Reply:
Just applied this with a new commit, fixing one reversal of the logic as we need the GNOME settings under wayland not under x11. Less loaded into RAM when running under x11 if we never load a schema or create a gsettings object that will never be used